controllers/managedocs_controller.go

```diff
@@ -607,11 +609,15 @@ func (r *ManagedOCSReconciler) reconcileStorageCluster() error {
 		// Handle only strict mode reconciliation
 		if r.reconcileStrategy == v1.ReconcileStrategyStrict {
 			// Get an instance of the desired state
-			desired := templates.StorageClusterTemplate.DeepCopy()
-			if err := r.updateStorageClusterFromAddonParamsSecret(desired); err != nil {
-				return err
+			var desired *ocsv1.StorageCluster = nil
+			var err error = nil
+			switch r.DeploymentType {
```

Comment:
Can we lowercase before switching? I would prefer for the value to be case insenstive